Spawned tasks cannot use the sm nor vader btl so you need an other one (tcp, openib, ...)<div>self btl is only to send/recvcount with oneself (e.g. it does not work for inter process and intra node communications.</div><div><br></div><div>I am pretty sure the lo interface is always discarded by openmpi, so I have no solution on top of my head that involves openmpi.</div><div>maybe your bed bet is to use a &quot;dummy&quot; interface, for example tan or tun or even a bridge.</div><div><br></div><div>Cheers,</div><div><br></div><div>Gilles</div><div><div><br></div><div><br><br>On Friday, March 11, 2016, Rémy Grünblatt &lt;<a href="mailto:remy.grunblatt@ens-lyon.fr">remy.grunblatt@ens-lyon.fr</a>&gt; wrote:<br><blockquote class="gmail_quote" style="margin:0 0 0 .8ex;border-left:1px #ccc solid;padding-left:1ex">Hello,<br>
I&#39;m having communications problem between two processes (with one being<br>
spawned by the other, on the *same* physical machine). Everything works<br>
as expected when I have network interface such as eth0 or wlo1 up, but<br>
as soon as they are down, I get errors (such as « At least one pair of<br>
MPI processes are unable to reach each other for MPI communications […] »).<br>
I tried to specify a set of mca parameters including the btl &quot;self&quot;<br>
parameter and including the lo interface in btl_tcp_if_include list, as<br>
advised by <a href="https://www.open-mpi.org/faq/?category=tcp" target="_blank">https://www.open-mpi.org/faq/?category=tcp</a> but I didn&#39;t reach<br>
any working state for this code with &quot;external&quot; network interface down.<br>
<br>
Got any idea about what I might do wrong ?<br>
<br>
Example code that triggers the problem: <a href="https://ptpb.pw/YOjr.tar.gz" target="_blank">https://ptpb.pw/YOjr.tar.gz</a><br>
Ompi_info:  <a href="https://ptpb.pw/Vt_V.txt" target="_blank">https://ptpb.pw/Vt_V.txt</a><br>
Full log: <a href="https://ptpb.pw/JCXn.txt" target="_blank">https://ptpb.pw/JCXn.txt</a><br>
<br>
Rémy<br>
<br>
<br>
</blockquote></div></div>

